src/main/java/mathandel/backend/service/RoleService.java

```diff
@@ -16,63 +19,67 @@
 import javax.validation.Valid;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+import static mathandel.backend.utils.ServerToClientDataConverter.mapModeratorRequest;
+
 @Service
 public class RoleService {
 
-    @Autowired
-    RoleRepository roleRepository;
-
-    @Autowired
-    ModeratorRequestsRepository moderatorRequestsRepository;
-
-    @Autowired
-    UserRepository userRepository;
-
-
-    public ApiResponse requestModerator(@Valid ModeratorRequestTO moderatorRequestTO, Long userId) {
-        Optional<User> optUser = userRepository.findById(userId);
-
-        if (optUser.isPresent() && hasRole(RoleName.ROLE_MODERATOR, optUser.get()))
-            throw  new BadRequestException("Role already given");
-
-        Optional<ModeratorRequest> optModeratorRequest = moderatorRequestsRepository.findByUser(optUser.get());
+    private final ModeratorRequestsRepository moderatorRequestsRepository;
+    private final UserRepository userRepository;
 
-        if (optModeratorRequest.isPresent())
-            throw  new BadRequestException("Request already submitted");
+    public RoleService(ModeratorRequestsRepository moderatorRequestsRepository, UserRepository userRepository) {
+        this.moderatorRequestsRepository = moderatorRequestsRepository;
+        this.userRepository = userRepository;
+    }
 
-        ModeratorRequest moderatorRequest = new ModeratorRequest();
+    public ApiResponse requestModerator( ModeratorRequestTO moderatorRequestTO, Long userId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new AppException("User doesn't exist"));
 
-        ModeratorRequestStatus moderatorRequestStatus = new ModeratorRequestStatus();
-        moderatorRequestStatus.setName(ModeratorRequestStatusName.PENDING);
+        if(moderatorRequestsRepository.existsByUser(user)) {
```

Comment:
also if request is rejected user can renew request so for example change the reason of request